opslevel/datasource_opslevel_domain.go

```diff
@@ -24,8 +24,39 @@ type DomainDataSource struct {
 	CommonDataSourceClient
 }
 
-// DomainDataSourceModel describes the data source data model.
-type DomainDataSourceModel struct {
+var domainDatasourceSchemaAttrs = map[string]schema.Attribute{
+	"id": schema.StringAttribute{
+		MarkdownDescription: "The ID of this Doamin",
+		Computed:            true,
+	},
+	"aliases": schema.ListAttribute{
+		ElementType: types.StringType,
+		Description: "The aliases of the domain.",
+		Computed:    true,
+	},
+	"name": schema.StringAttribute{
+		Description: "The name of the domain.",
+		Computed:    true,
+	},
+	"description": schema.StringAttribute{
+		Description: "The description of the domain.",
+		Computed:    true,
+	},
+	"owner": schema.StringAttribute{
+		Description: "The id of the domain owner - could be a group or team.",
+		Computed:    true,
+	},
+}
+
+func DomainAttributes(attrs map[string]schema.Attribute) map[string]schema.Attribute {
```

Comment:
Confused - what does this do?

Reply:
This is a factory function that appends `attrs` (extra schema attributes) to `domainDatasourceSchemaAttrs` (base/shared domain datasource schema attributes). 

This lets us reuse `domainDatasourceSchemaAttrs` here for `DomainDataSource` and in `opslevel/datasource_opslevel_domains_all.go` for `DomainDataSourcesAll`. `DomainDataSource` has the extra `"identifier": schema.StringAttribute{}` attribute but the two are otherwise the same